File: java/src/main/java/gov/doc/ntia/sigmf/examples/location/SimpleGCS.java
package gov.doc.ntia.sigmf.examples.location;

import gov.doc.ntia.sigmf.Capture;
import gov.doc.ntia.sigmf.Global;
import gov.doc.ntia.sigmf.MetaDoc;
import gov.doc.ntia.sigmf.examples.Example;
import gov.doc.ntia.sigmf.examples.ExampleUtils;
import gov.doc.ntia.sigmf.ext.global.location.GeographicCoordinateSystem;

import java.util.Calendar;
import java.util.Date;
import java.util.List;

public class SimpleGCS implements Example {
    @Override
    public MetaDoc getExample() {
        Date now = Calendar.getInstance().getTime();
        MetaDoc metadoc = new MetaDoc();
        Global global = ExampleUtils.getGlobal(now);
        List<Capture> captures = ExampleUtils.getCaptures(now);
        metadoc.setGlobal(global);
        metadoc.setCaptures(captures);
        GeographicCoordinateSystem gcs = new GeographicCoordinateSystem();
        gcs.setId("EPSG:4326");
        global.setCoordinateSystem(gcs);
        return metadoc;
    }
}

File: C#/sigmf/sigmf/examples/location/SimpleGCS.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf.examples.location
{
    public class SimpleGCS : Example
    {
        public override MetaDoc getExample()
        {
            DateTime now = getCurrentTime();
            MetaDoc metaDoc = new MetaDoc();
            Globals globals = ExampleUtils.getGlobals(now);
            List<Capture> captures = ExampleUtils.getCaptures(now);
            metaDoc.setGlobal(globals);
            metaDoc.setCaptures(captures);
            GeographicCoordinateSystem gcs = new GeographicCoordinateSystem();
            gcs.Id = "EPSG:4326";
            globals.CoordinateSystem = gcs;
            return metaDoc;
        }
        private DateTime getCurrentTime()
        {
            DateTime st = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
            TimeSpan t = (DateTime.Now.ToUniversalTime() - st);
            return st;
        }
    }
}
